salt/client/mixins.py

```diff
@@ -72,7 +72,7 @@ def wrapper(*args, **kwargs):
                    }
             pub_data = {}
             # pull out pub_data if you have it
-            for k, v in six.iteritems(list(kwargs.items())):
+            for k, v in six.iteritems(dict(kwargs)):
```

Comment:
To make a copy of the dict, before this was doing .items() which iterates over a *copy* of the dict